assets/prometheus-k8s/rules.yaml

```diff
@@ -1338,11 +1338,12 @@ spec:
         severity: critical
     - alert: etcdHighNumberOfLeaderChanges
       annotations:
-        message: 'etcd cluster "{{ $labels.job }}": instance {{ $labels.instance }}
-          has seen {{ $value }} leader changes within the last 30 minutes.'
+        message: 'etcd cluster "{{ $labels.job }}": {{ $value }} leader changes within
+          the last 15 minutes. Frequent elections may be a sign of insufficient resources,
+          high network latency, or disruptions by other components and should be investigated.'
       expr: |
-        rate(etcd_server_leader_changes_seen_total{job=~".*etcd.*"}[15m]) > 3
-      for: 15m
+        increase((max by (job) (etcd_server_leader_changes_seen_total{job=~".*etcd.*"}) or 0*absent(etcd_server_leader_changes_seen_total{job=~".*etcd.*"}))[15m:1m]) >= 3
```

Comment:
why is `== 3` bad? With 3 masters and a rolling upgrade of the nodes, couldn't you get 3 leader changes?